In [23]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split

df = pd.read_csv('../data/data.csv').dropna()

print df.shape
print df.columns

(25697, 25)
Index([u'action_type', u'combined_shot_type', u'game_event_id', u'game_id',
       u'lat', u'loc_x', u'loc_y', u'lon', u'minutes_remaining', u'period',
       u'playoffs', u'season', u'seconds_remaining', u'shot_distance',
       u'shot_made_flag', u'shot_type', u'shot_zone_area', u'shot_zone_basic',
       u'shot_zone_range', u'team_id', u'team_name', u'game_date', u'matchup',
       u'opponent', u'shot_id'],
      dtype='object')


In [24]:
df.head()

action_type combined_shot_type  game_event_id   game_id      lat  \
1          Jump Shot          Jump Shot             12  20000012  34.0443   
2          Jump Shot          Jump Shot             35  20000012  33.9093   
3          Jump Shot          Jump Shot             43  20000012  33.8693   
4  Driving Dunk Shot               Dunk            155  20000012  34.0443   
5          Jump Shot          Jump Shot            244  20000012  34.0553   

   loc_x  loc_y       lon  minutes_remaining  period   ...          shot_type  \
1   -157      0 -118.4268                 10       1   ...     2PT Field Goal   
2   -101    135 -118.3708                  7       1   ...     2PT Field Goal   
3    138    175 -118.1318                  6       1   ...     2PT Field Goal   
4      0      0 -118.2698                  6       2   ...     2PT Field Goal   
5   -145    -11 -118.4148                  9       3   ...     2PT Field Goal   

          shot_zone_area  shot_zone_basic  shot_zone_range     team_id  \
1           Left Side(L)        Mid-Range         8-16 ft.  1610612747   
2   Left Side Center(LC)        Mid-Range        16-24 ft.  1610612747   
3  Right Side Center(RC)        Mid-Range        16-24 ft.  1610612747   
4              Center(C)  Restricted Area  Less Than 8 ft.  1610612747   
5           Left Side(L)        Mid-Range         8-16 ft.  1610612747   

            team_name   game_date    matchup opponent  shot_id  
1  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        2  
2  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        3  
3  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        4  
4  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        5  
5  Los Angeles Lakers  2000-10-31  LAL @ POR      POR        6  

[5 rows x 25 columns]

## Information about the data
As we can see, there are 25,697 labelled shots and 25 columns (one column is what we're trying to predict, which is whether or not he made the shot). Here are brief descriptions of what each column means:

- action_type:

Values: Jump Shot, Driving Dunk, Layup Shot, Running Jump Shot, etc.

This basically means the type of shot that Kobe took


- combined_shot_type

Values: Jump Shot, Dunk, Layup

- game_event_id

- game_id

- lat

- loc_x

- loc_y

- lon

- minutes_remaining

- period

- playoffs

- season

- seconds_remaining

- shot_distance

- shot_made_flag

- shot_type

- shot_zone_area

- shot_zone_basic

- shot_zone_range

- team_id

- team_name

- game_date

- matchup

- opponent

- shot_id


In [19]:
drop_list = ['combined_shot_type', 'game_event_id', 'game_id', 'lat', 'lon', 'shot_type', 'shot_zone_area',
            'shot_zone_basic', 'shot_zone_range', 'team_id', 'team_name', 'shot_id', 'opponent']

def drop_columns(df):
    return df.drop(drop_list, axis=1)

df = drop_columns(df)
print df.shape

(25697, 12)


As you can see, dropping these redundant 

In [13]:
y = df['shot_made_flag']
X = df.drop('shot_made_flag', axis=1)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)